src/extension.ts

```diff
@@ -22,14 +23,16 @@ import { LoadedScriptsProvider, openScript } from './explorer/loadedScriptsExplo
 import { getAppServicePlan, SiteTreeItem } from './explorer/SiteTreeItem';
 import { WebAppProvider } from './explorer/WebAppProvider';
 import { WebAppTreeItem } from './explorer/WebAppTreeItem';
-import { Reporter, reporter } from './telemetry/reporter';
+import { getPackageInfo, IPackageInfo } from './telemetry/reporter';
 import * as util from "./util";
 import { nodeUtils } from './utils/nodeUtils';
 
 // tslint:disable-next-line:export-name
 // tslint:disable-next-line:max-func-body-length
 export function activate(context: vscode.ExtensionContext): void {
-    context.subscriptions.push(new Reporter(context));
+    const packageInfo: IPackageInfo = getPackageInfo(context);
+    const reporter = new TelemetryReporter(packageInfo.name, packageInfo.version, packageInfo.aiKey);
+    context.subscriptions.push(reporter);
```

Comment:
Just curious, was this change mainly because the Reporter class was used just once?

Reply:
Yeah, we used to have the wrapper handle telemetry, but now it's handled through the AzureActionHandler when we register the commands.